internal/xds/translator/cluster.go

```diff
@@ -77,6 +78,30 @@ func buildXdsCluster(args *xdsClusterArgs) *clusterv3.Cluster {
 		cluster.TransportSocket = args.tSocket
 	}
 
+	for i, ds := range args.settings {
+		if ds.TLS != nil {
+			// for upstreamTLS , a fixed sni can be used. use auto_sni otherwise
+			// https://www.envoyproxy.io/docs/envoy/latest/faq/configuration/sni#faq-how-to-setup-sni:~:text=For%20clusters%2C%20a,for%20trust%20anchor.
+			socket, err := buildXdsUpstreamTLSSocketWthCert(ds.TLS)
+			if err != nil {
+				// TODO: Log something here
+				return nil
+			}
+			if args.proxyProtocol != nil {
+				socket = buildProxyProtocolSocket(args.proxyProtocol, socket)
+			}
+			cluster.TransportSocketMatches = append(cluster.TransportSocketMatches, &clusterv3.Cluster_TransportSocketMatch{
+				Name: fmt.Sprintf("destination-tls-config-%d", i),
```

Comment:
nit: each setting has a 1:1 mapping with a k8s backendRef
we could enhance https://github.com/envoyproxy/gateway/blob/f595862795047ddc35a2d181e8419f68af21187e/internal/ir/xds.go#L819 to add a `name` field and use that here in the metadata as `name: <nsName>`
(can be a big diff, so non blocker)

Reply:
I changed the name so that it's based on the cluster-name and is a little easier to understand.

Adding the name to the destination settings is a bigger change, since these are created in all sorts of places. Maybe a future PR?